open_spiel/games/universal_poker.cc

```diff
@@ -1091,10 +1094,16 @@ int UniversalPokerGame::MaxGameLength() const {
     maxBlind =
         acpc_game_.BlindSize(p) > maxBlind ? acpc_game_.BlindSize(p) : maxBlind;
   }
-
-  while (maxStack > maxBlind) {
-    maxStack /= 2.0;         // You have always to bet the pot size
-    length += NumPlayers();  // Each player has to react
+  if ((betting_abstraction_==BettingAbstraction::kFULLGAME) || (betting_abstraction_==BettingAbstraction::kFCHPA)){
+    // with fullgame, the longest game comes from each player can bet/raise the big blind every action.
+    // with FCHPA, the longest game is when each player bets/raise half-pot every action.
+    // however, for now we'll just use the fullgame value for FCHPA too, although it is a big overestimate.
+    length += (maxStack+maxBlind-1)/maxBlind;
+  } else {
+    while (maxStack > maxBlind) {
+      maxStack /= 2.0;             // You have always to bet the pot size
+      length += NumPlayers() - 1;  // 1 player bets, and n-2 players call
```

Comment:
Why is it n-2 players call and not n-1?

Reply:
Good question. Here's my thought process. Also let me know if you see anything wrong with the reasoning.

The longest game (with a single betting round, for simplicity) consists of:

n-1 players checking,  
1 player betting, n-2 players calling,  
1 player raising, n-2 players calling,  
etc...,  
1 player raising, n-1 players calling

so each min-bet/min-raise is succeeded by n-2 calls (except for the last bet, which is succeeded by n-1). In addition, there are n-1 checks in the beginning. These n-1 checks + the final 1 call add up to n, which is accounted for a few lines above in the code (`// Check Actions`).

Reply:
Actually it's probably convenient for future readers of the code if I just put this comment as a code comment. What do ya think?

Reply:
> First, the calculation on line 1101 only accounts for the number of raises, not all of the n-2 calls in between. So I believe that has to be updated.

ah yes, thanks. 😅 good catch! 

> I think this is more intuitive than dividing the max stack size in half.

Yup, definitely. The logic in your snippet all looks good, thanks!

> Funny how even these simple things can be tricky!

On the bright side, this has finally made me learn how to calculate a pot-sized bet.